apps/src/lib/node/ledger/shell/vote_extensions/bridge_pool_vext.rs

```diff
@@ -88,20 +85,35 @@ where
         {
             Some(epoch) => epoch,
             _ => {
-                tracing::error!(
+                tracing::debug!(
                     block_height = ?ext.data.block_height,
                     "The epoch of the Bridge pool root's vote extension's \
                      block height should always be known",
                 );
                 return Err(VoteExtensionError::UnexpectedEpoch);
             }
         };
+        if !self
```

Comment:
This debug line should come first. It would be very strange to get one of the above errors if we expect the bridge to be inactive.

Reply:
I kept the block height query before the epoch and bridge queries on purpose. this is because it's very cheap to read a value from memory and compare it against another value in memory. checking the epoch doesn't need to read a value from storage either, but it's slightly more expensive, because it takes linear time, based on the number of epochs we've gone through. checking the status of the bridge does read a value from storage, so it would be easy to abuse it to send invalid protocol txs.

idk, I guess we need to read values from storage anyway, to confirm who signed the protocol tx, etc. what do you think?

Reply:
I'm going to err on the side of better (or more illuminating and expected) error messages. These kinds of small optimizations are not as important. Blockchains aren't fast.